### PR TITLE
Add ACS2 with ER extension

### PR DESCRIPTION
**Pull request by GodspeedYouBlackEmperor**
Add ACS2 with ER extension

### CODE REVIEW

File: lcs/agents/acs2er/ACS2ER.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+class ACS2ER(Agent):
+
+    def __init__(self,
+                 cfg: Configuration,
+                 population: ClassifiersList = None,
+                 replay_memory: ReplayMemory = None) -> None:
```
Comment (khozzy) [inline lcs/agents/acs2er/ACS2ER.py:20]: I'd rather prefer to keep agent constructors taking only the `cfg` and `population` parameters. The replay memory size can later be fetched from `self.cfg`

File: lcs/agents/acs2er/ReplayMemory.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    """
+    Represents the replay memory buffer
+    """
+
+    def __init__(self, *args, max_size: int, oktypes=(ReplayMemorySample,)) -> None:
+        super().__init__(*args, oktypes=oktypes)
+        self.max_size = max_size
+
+    def update(self, sample: ReplayMemorySample) -> None:
+        if(len(self) >= self.max_size):
```
Comment (khozzy) [inline lcs/agents/acs2er/ReplayMemory.py:17]: No need for parenthesis

File: lcs/agents/acs2er/ReplayMemorySample.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from __future__ import annotations
+from lcs import Perception
+
+
+class ReplayMemorySample:
```
Comment (khozzy) [inline lcs/agents/acs2er/ReplayMemorySample.py:5]: Maybe a `@dataclass`?

File: lcs/agents/acs2er/ACS2ER.py (hunk 1 above)
Comment (GodspeedYouBlackEmperor) [reply lcs/agents/acs2er/ACS2ER.py]: done

File: lcs/agents/acs2er/ReplayMemory.py (hunk 2 above)
Comment (GodspeedYouBlackEmperor) [reply lcs/agents/acs2er/ReplayMemory.py]: done

File: lcs/agents/acs2er/ReplayMemorySample.py (hunk 3 above)
Comment (GodspeedYouBlackEmperor) [reply lcs/agents/acs2er/ReplayMemorySample.py:6]: done

**Pull request merged**